Severity from buffer

kirubaspace opened the pull request:
This fixes https://github.com/goatshriek/stumpless/issues/383 

Make sure that your pull request follows the guidelines specified in the
[Guidelines for Contributing](../docs/CONTRIBUTING.md).

kirubaspace committed:
Initial changes

kirubaspace committed:
Initial changes for buffer

kirubaspace committed:
Removed spaces

kirubaspace committed:
Added more UT

kirubaspace commented:
@goatshriek Could  you please review when you get time

goatshriek commented:
There are merge conflicts that must be resolved before this will be reviewed. It looks like you will need to do a fresh pull from `latest`, as the functionality has changed a bit since this branch appears to have started.

kirubaspace committed:
Fix merge conflicts

goatshriek reviewed (changes requested):
Please review the current `stumpless_get_severity_enum` functionality and tests - it has been expanded recently. You'll need to make sure these newer features are preserved.

Rather than simply copying the functionality directly, it will make more sense to move the functionality to your function, and have the previous function call your new `_from_buffer` one so that code is not duplicated.

kirubaspace committed:
Addressed review comments

kirubaspace commented:
> Please review the current `stumpless_get_severity_enum` functionality and tests - it has been expanded recently. You'll need to make sure these newer features are preserved.
> 
> Rather than simply copying the functionality directly, it will make more sense to move the functionality to your function, and have the previous function call your new `_from_buffer` one so that code is not duplicated.

Moved the functionality to new function and updated the unit test. Please let me know if this is fine or any changes required

kirubaspace committed:
Removed spaces

kirubaspace committed:
Removed spaces

goatshriek commented on test/function/severity.cpp:
```diff
@@ -122,4 +122,24 @@ namespace {
+    ASSERT_NOT_NULL( set_malloc_result );
+
+    result = stumpless_get_severity_enum_from_buffer( "info", sizeof("info") );
+    EXPECT_EQ( result, -1 );
```
There should also be an `EXPECT_ERROR_ID_EQ` check here to make sure that the error code is what is expected in this case. It looks like that's also missing in the test that you based this one on, so feel free to add it there as well.

goatshriek reviewed (changes requested):
Thanks for putting this together and getting it updated so quickly! Just a few minor tweaks for you to make.

To resolve the Windows errors, you'll need to add your new function to `src/windows/stumpless.def`, which will expose the call to linkers.

kirubaspace committed:
Addressed review comments
